src/lib/addons/slack-app.test.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    FEATURE_CREATED,
+    FEATURE_ARCHIVED,
+    FEATURE_ENVIRONMENT_DISABLED,
+    IEvent,
+} from '../types/events';
+
+import SlackAppAddon from './slack-app';
+
+import noLogger from '../../test/fixtures/no-logger';
+import { ChatPostMessageArguments } from '@slack/web-api';
+
+const accessToken = 'test-access-token';
+const slackApiCalls: ChatPostMessageArguments[] = [];
+jest.mock('@slack/web-api', () => ({
+    WebClient: jest.fn().mockImplementation(() => ({
+        conversations: {
+            list: () => ({
+                channels: [
+                    {
+                        id: 1,
+                        name: 'general',
+                    },
+                    {
+                        id: 2,
+                        name: 'another-channel-1',
+                    },
+                ],
+            }),
+        },
+        chat: {
+            postMessage: jest.fn().mockImplementation((options) => {
+                slackApiCalls.push(options);
+                return Promise.resolve();
+            }),
+        },
+    })),
+}));
+
+beforeEach(() => {
+    slackApiCalls.length = 0;
+});
+
+test('Should call Slack API', async () => {
+    const addon = new SlackAppAddon({
+        getLogger: noLogger,
+        unleashUrl: 'http://some-url.com',
+    });
+    const event: IEvent = {
+        id: 1,
+        createdAt: new Date(),
+        type: FEATURE_CREATED,
+        createdBy: 'some@user.com',
+        project: 'default',
+        featureName: 'some-toggle',
+        data: {
+            name: 'some-toggle',
+            enabled: false,
+            type: 'release',
+            strategies: [{ name: 'default' }],
+        },
+        tags: [{ type: 'slack', value: 'general' }],
+    };
+
+    await addon.handleEvent(event, { accessToken });
+    expect(slackApiCalls.length).toBe(1);
+    expect(slackApiCalls[0].channel).toBe(1);
+    expect(slackApiCalls[0]).toMatchSnapshot();
+});
+
+test('Should call Slack API for archived toggle', async () => {
+    const addon = new SlackAppAddon({
+        getLogger: noLogger,
+        unleashUrl: 'http://some-url.com',
+    });
+    const event: IEvent = {
+        id: 2,
+        createdAt: new Date(),
+        type: FEATURE_ARCHIVED,
+        createdBy: 'some@user.com',
+        project: 'default',
+        featureName: 'some-toggle',
+        data: {
+            name: 'some-toggle',
+        },
+        tags: [{ type: 'slack', value: 'general' }],
+    };
+
+    await addon.handleEvent(event, { accessToken });
+    expect(slackApiCalls.length).toBe(1);
+    expect(slackApiCalls[0].channel).toBe(1);
+    expect(slackApiCalls[0]).toMatchSnapshot();
+});
```

Comment:
This is probably not needed, AFAIK the logic to deliver an event to an addon is elsewhere. SlackAppAddon doesn't have any rejection logic. Doesn't hurt to have more tests though, but just mentioning it :D

Reply:
Yep, but to be fair these tests are there in the old slack addon. I guess they are there simply for the extra snapshots. I can remove these for now.

Reply:
Addressed in https://github.com/Unleash/unleash/pull/4259/commits/34f9b6c4f26d8c885bfc177e4a4ab3d67b5cac42